### PR TITLE
[FLINK-34282][ci] Adds release-1.18 back to GHA nightly trigger workflow

### PR DESCRIPTION
**Pull request by XComp**
## What is the purpose of the change

https://github.com/apache/flink/commit/b8b2596a updated the release branch to monitor from `release-1.18` to `release-1.19` because of the 1.19 release branch cut. We still want to monitor `release-1.18` as part of the nightlies, though.

## Brief change log

* Adds `release-1.18` to the branch list for the GHA nightly trigger workflow

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? no
  - If yes, how is the feature documented? not applicable

### CODE REVIEW

**Review by lincoln-lil — APPROVED** [review]
+1 for monitor more versions via the new GHA workflow

**Comment by XComp** [comment]
Thanks for the approval, @lincoln-lil . I'm not gonna wait for CI for that one because it's a configuration change unrelated to the Azure Pipelines.

**Comment by lincoln-lil** [comment]
@XComp  This was my oversight, it should be the recent two releases + master, not two branches in total, thanks for the fixing :)

**Comment by XComp** [comment]
> @XComp This was my oversight, it should be the recent two releases + master, not two branches in total, thanks for the fixing :)

No worries. I'm happy to help. :-) 

But on that note: The idea of moving the release process into Jira issues was to make the release process reviewable and have each of the steps being associated with a Jira issue. Release-related commits can be also assigned to the corresponding Jira. That helps tracing changes back through the git history.

Creating PRs for each Jira and/or documenting changes in the Jira comment section would help prevent (or at least reduce) these kind of issues because it allows for a second pair of eyes to look over the change. There is a list of expected outcomes for each of the Jiras which a reviewer could follow. :-)

**Comment by lincoln-lil** [comment]
Thank you very much for the detailed explanation, I'll follow the rules for subsequent operations to avoid similar problems.

**Pull request merged**
